python/cuml/manifold/umap.pyx

```diff
@@ -371,44 +381,38 @@ class UMAP(Base):
             ndarray, cuda array interface compliant array like CuPy
         """
 
-        if self._should_downcast:
-            warnings.warn("Parameter should_downcast is deprecated, use "
-                          "convert_dtype in fit, fit_transform and transform "
-                          " methods instead. ")
-            convert_dtype = True
-
         if len(X.shape) != 2:
             raise ValueError("data should be two dimensional")
 
-        self.X_m, X_ctype, n_rows, n_cols, dtype = \
+        self.X_m, X_ctype, self.n_rows, self.n_dims, dtype = \
             input_to_dev_array(X, order='C', check_dtype=np.float32,
                                convert_to_dtype=(np.float32 if convert_dtype
                                                  else None))
 
-        if n_rows <= 1:
+        if self.n_rows <= 1:
             raise ValueError("There needs to be more than 1 sample to "
                              "build nearest the neighbors graph")
 
         cdef UMAPParams * umap_params = \
-            <UMAPParams*> < size_t > self.umap_params
-        umap_params.n_neighbors = min(n_rows, umap_params.n_neighbors)
-        self.n_dims = n_cols
-        self.raw_data = X_ctype
-        self.raw_data_rows = n_rows
+            <UMAPParams*> <size_t> self.umap_params
+        umap_params.n_neighbors = min(self.n_rows, umap_params.n_neighbors)
 
-        self.arr_embed = rmm.to_device(zeros((self.X_m.shape[0],
+        self.embedding_ = rmm.to_device(zeros((self.n_rows,
                                               umap_params.n_components),
-                                             order="C", dtype=np.float32))
-        self.embeddings = \
-            self.arr_embed.device_ctypes_pointer.value
+                                              order="C", dtype=np.float32))
+
+        self.input_hash = joblib.hash(self.X_m.copy_to_host())
```

Comment:
Could you explain a bit why we need the input hash? Seems like it would be nice to avoid copying the whole dataset to cpu memory to hash it if we can avoid that.